docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -348,6 +348,310 @@ and a file system location of ``/var/my_tables/test_table``::
+
+.. list-table:: Partitions columns
+  :widths: 20, 30, 50
+  :header-rows: 1
+
+  * - Name
+    - Type
+    - Description
+  * - ``partition``
+    - ``row(...)``
+    - A row which contains the mapping of the partition column name(s) to the partition column value(s)
+  * - ``record_count``
+    - ``bigint``
+    - The number of records under the partition
+  * - ``file_count``
+    - ``bigint``
+    - The number of files mapped under the partition
+  * - ``total_size``
+    - ``bigint``
+    - The size of all the files under the partition
+  * - ``data``
+    - ``row(... row (min ..., max ... , null_count bigint))``
+    - Partition range metadata
+
+``$files`` table
+^^^^^^^^^^^^^^^^
+
+The ``$files`` table provides a detailed overview of the data files corresponding to the  Iceberg table.
+
+To retrieve the information about the data files of the Iceberg table ``test_table`` use the query::
+
+    SELECT * FROM "test_table$files"
+
+.. code-block:: text
+
+     content  | file_path                                                                                                                     | record_count    | file_format   | file_size_in_bytes   |  column_sizes        |  value_counts     |  null_value_counts | nan_value_counts  | lower_bounds                |  upper_bounds               |  key_metadata  | split_offsets  |  equality_ids
+    ----------+-------------------------------------------------------------------------------------------------------------------------------+-----------------+---------------+----------------------+----------------------+-------------------+--------------------+-------------------+-----------------------------+-----------------------------+----------------+----------------+---------------
+     0        | hdfs://hadoop-master:9000/user/hive/warehouse/test_table/data/c1=3/c2=2021-01-14/af9872b2-40f3-428f-9c87-186d2750d84e.parquet |  1              |  PARQUET      |  442                 | {1=40, 2=40, 3=44}   |  {1=1, 2=1, 3=1}  |  {1=0, 2=0, 3=0}   | <null>            |  {1=3, 2=2021-01-14, 3=1.3} |  {1=3, 2=2021-01-14, 3=1.3} |  <null>        | <null>         |   <null>
+
+
+
+The output of the query has the following columns:
+
+.. list-table:: Files columns
+  :widths: 20, 30, 50
+  :header-rows: 1
+
+  * - Name
+    - Type
+    - Description
+  * - ``content``
+    - ``integer``
+    - Type of content stored in the file.
+      The supported types of content supported in Iceberg are
+
+      * ``DATA(0)``
+      * ``POSITION_DELETES(1)``
+      * ``EQUALITY_DELETES(2)``
+  * - ``file_path``
+    - ``varchar``
+    - The data file location
+  * - ``file_format``
+    - ``varchar``
+    - The format of the data file
+  * - ``record_count``
+    - ``bigint``
+    - The number of entries contained in the data file
+  * - ``file_size_in_bytes``
+    - ``bigint``
+    - The data file size
+  * - ``column_sizes``
+    - ``map(integer, bigint)``
+    - The size for each column in the columnar file
```

Comment:
Nit: Would it be worth mentioning for these that the integer keys are the field IDs used by Iceberg?

I don't have a good way at the moment to express that (especially without looking at the rest of the existing docs). This could be a follow up item, if anything.

Reply:
Good point.

I have adapted this to:

>  Mapping between the Iceberg column ID and its corresponding size within the columnar file